### PR TITLE
lapack/gonum: add Dlauu2 and Dlauum

### PR DESCRIPTION
**Pull request by vladimir-ch**
Please take a look.

Updates #746

### CODE REVIEW

File: lapack/testlapack/dlauu2.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+			}
+
+			// * Check that the triangle opposite to uplo has not been modified.
+			// * Convert the result of Dlauu? into a dense symmetric matrix.
+			// * Zero out the triangle in aCopy opposite to uplo.
+			if uplo == blas.Upper {
+				for i := 1; i < n; i++ {
+					for j := 0; j < i; j++ {
+						if aCopy[i*lda+j] != a[i*lda+j] {
+							t.Errorf("%v: unexpected modification in upper triangle", prefix)
```
Comment (kortschak) [inline lapack/testlapack/dlauu2.go]: lower?

File: lapack/testlapack/dlauu2.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+						}
+						a[i*lda+j] = a[j*lda+i]
+						aCopy[i*lda+j] = 0
+					}
+				}
+			} else {
+				for i := 0; i < n-1; i++ {
+					for j := i + 1; j < n; j++ {
+						if aCopy[i*lda+j] != a[i*lda+j] {
+							t.Errorf("%v: unexpected modification in lower triangle", prefix)
```
Comment (kortschak) [inline lapack/testlapack/dlauu2.go]: upper?

File: lapack/testlapack/dlauu2.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"fmt"
+	"testing"
+
+	"golang.org/x/exp/rand"
```
Comment (kortschak) [inline lapack/testlapack/dlauu2.go]: Blank line between x/exp import and gonum imports.

**Comment by vladimir-ch** [comment]
The bug in `GemvT` was unexpected but it is fixed now.

**Comment by kortschak** [comment]
Nice bug finding.

File: lapack/testlapack/dlauu2.go (hunk 1 above)
Comment (vladimir-ch) [reply lapack/testlapack/dlauu2.go]: I had it correct here, then copy/pasted below, then noticed I had "lower" twice and fixed the wrong one to upper.

Done.

File: lapack/testlapack/dlauu2.go (hunk 2 above)
Comment (vladimir-ch) [reply lapack/testlapack/dlauu2.go]: Done.

File: lapack/testlapack/dlauu2.go (hunk 3 above)
Comment (vladimir-ch) [reply lapack/testlapack/dlauu2.go]: Done.

**Pull request merged**